### PR TITLE
fix version of ng-diff-match-patch (2.0.6 to 3.0.1) for angular 9

### PR DESCRIPTION
**Pull request by jbacquet**
ng-diff-match-patch v.2.0.6 is not compatible with angular 9 Ivy and needs to be upgraded to latest version  3.0.1

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "generator-jhipster-entity-audit",
-  "version": "3.3.0",
+  "version": "3.3.1",
```
Comment (murdos) [inline package.json]: This will be done automatically during the release. Thus could you revert this change? Thanks :)

File: generators/app/index.js [hunk 2]
```diff
@@ -418,7 +418,7 @@ module.exports = class extends BaseGenerator {
         if (!this.skipClient) {
           if (this.clientFramework === 'angularX') {
             // add dependency required for displaying diffs
-            this.addNpmDependency('ng-diff-match-patch', '2.0.6');
+            this.addNpmDependency('ng-diff-match-patch', '3.0.1');
```
Comment (murdos) [inline generators/app/index.js]: Where does this version comes from? https://github.com/elliotforbes/ng-diff-match-patch only contains 2.0.6 version

Comment (jbacquet) [reply generators/app/index.js:421]: Sorry for the late response, I didn't receive notification. How do you want to me to do that (revert only this changes) ? To be honest, it's my first pull request :)

File: package.json (hunk 1 above)
Comment (jbacquet) [reply package.json:3]: Sorry for the late response, I didn't receive notification. How do you want to me to do that (revert only this changes) ? To be honest, it's my first pull request :)

File: generators/app/index.js (hunk 2 above)
Comment (jbacquet) [reply generators/app/index.js:421]: This version comes directly from [NPM](https://www.npmjs.com/package/ng-diff-match-patch/v/3.0.1). An issue where opened to knows where come from this version 3.0.1 [here](
https://github.com/elliotforbes/ng-diff-match-patch/issues/39 )

File: package.json (hunk 1 above)
Comment (murdos) [reply package.json:3]: You can make a new commit that changes the version number back, and ideally squash it with the previous commit.

File: generators/app/index.js (hunk 2 above)
Comment (murdos) [reply generators/app/index.js:421]: OK. I'd rather wait to know the origin of this npm release.

**Comment by DanielFran** [comment]
Issue confirmed!
@murdos Tested with generator v6.10.4 and this update is needed

**Comment by murdos** [comment]
> 
> 
> Issue confirmed!
> @murdos Tested with generator v6.10.4 and this update is needed

I agree it's needed. But the problem is that the source of version 3.0.1 are not public and thus we don't know what it contains. It might be compromised, that I'd rather not do the update.

**Comment by DanielFran** [comment]
@murdos The sources are available: https://github.com/elliotforbes/ng-diff-match-patch/compare/2.0.6...master

Current version 3.0.1 contains probably commits until Jul 03, 2018
